src/tests/ftest/deployment/target_failure.py

```diff
@@ -0,0 +1,417 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import time
+import os
+import threading
+
+from ior_test_base import IorTestBase
+from ior_utils import IorCommand
+from general_utils import report_errors
+from command_utils_base import CommandFailure
+from job_manager_utils import get_job_manager
+
+
+class TargetFailure(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description: Verify target failure is properly handled and recovered.
+
+    :avocado: recursive
+    """
+    def run_ior_report_error(self, results, job_num, file_name, oclass, pool, container):
+        """Run IOR command and store the results to results dictionary.
+
+        Create a new IorCommand object instead of using the one in IorTestBase because
+        we'll run a test that runs multiple IOR processes at the same time.
+
+        Args:
+            results (dict): A dictionary object to store the ior metrics
+            job_num (int): Assigned job number
+            file_name (str): File name used for self.ior_cmd.test_file.
+            oclass (str): Value for dfs_oclass and dfs_dir_oclass.
+            pool (TestPool): Pool to run IOR.
+            container (TestContainer): Container to run IOR.
+        """
+        # Update the object class depending on the test case.
+        ior_cmd = IorCommand()
+        ior_cmd.get_params(self)
+        ior_cmd.dfs_oclass.update(oclass)
+        ior_cmd.dfs_dir_oclass.update(oclass)
```

Comment:
FYI, you could also make use of yaml inheritance and use multiple IOR namespaces:
https://github.com/daos-stack/daos/blob/7fe4882dea479542fd2f3c11c902746249d8a4d9/src/tests/ftest/performance/ior_easy.yaml#L55-L75

Reply:
Done. Thanks for the trick.